src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -203,6 +215,14 @@ internal static unsafe void Initialize (JnienvInitializeArgs* args)
 			SynchronizationContext.SetSynchronizationContext (Android.App.Application.SynchronizationContext);
 #endif
 
+#if NET
+			if (args->mappingXml != IntPtr.Zero) {
+				var xml = Encoding.UTF8.GetString ((byte*) args->mappingXml, args->mappingXmlLen);
+				Logger.Log (LogLevel.Warn, "*jonp*", $"# jonp: mapping xml: len={args->mappingXmlLen}; {xml}");
+				(ReplacementTypes, ReplacementMethods) = MamXmlParser.ParseStrings (xml);
```

Comment:
Instead of using XML here at all, could we generate a type in the main assembly that has this data? So you would just use reflection to look for a couple well-known class and field names?

Maybe that is more work than we're willing to do right now.

Reply:
See my [other comment](https://github.com/xamarin/xamarin-android/pull/6591#discussion_r870607684), but…. We don't *want* to "privilege" the "main" assembly ("what's a main assembly?"), because when we *have*, things *break*; see e.g. 807e66501257d2adee731df1ed501301e2083530, wherein an app crashed because the Activity launched *wasn't* in the "main" assembly.

The "best" fix is to put this data into `libxamarin-app.so`.  I think we can do that by e.g. .NET 7 P7.